cmake/googletest.cmake

```diff
@@ -0,0 +1,17 @@
+# Google Test library
+#
+# Google Test is a testing framework for C++ developed by Google.
+#
+# This CMake script integrates Google Test into the project, allowing easy usage.
+# Usage example: target_link_libraries(my_target PRIVATE GTest::GTest)
+
+include(FetchContent)
+
+FetchContent_Declare(
+  googletest
+  GIT_REPOSITORY https://github.com/google/googletest.git
+  GIT_TAG        release-1.12.1
```

Comment:
It there any reason not to use current version `v1.14.0`?

Reply:
Updated to newer version